homeassistant/loader.py

```diff
@@ -541,25 +521,15 @@ async def async_get_integration(hass: HomeAssistant, domain: str) -> Integration
     integration = await hass.async_add_executor_job(
         Integration.resolve_from_root, hass, components, domain
     )
-
-    if integration is not None:
-        cache[domain] = integration
-        event.set()
-        return integration
-
-    integration = Integration.resolve_legacy(hass, domain)
```

Comment:
We didn't explicitly mention the requirement for the custom integration having a manifest.json but I guess that is implied from it requiring a version key?

Reply:
We have done that in every breaking change note since 2021.2

Reply:
Good! I found it here at least:
https://www.home-assistant.io/blog/2021/03/03/release-20213/#breaking-changes

Reply:
Yeah, I was wrong, 2021.2 only adjusted hassfest, 2021.3/4/5 all have the same note